src/axom/primal/geometry/BoundingBox.hpp

```diff
@@ -410,11 +420,10 @@ bool BoundingBox<T, NDIMS>::intersectsWith(
   // AABBs cannot intersect if they are separated along any dimension
   for(int i = 0; i < NDIMS; ++i)
   {
-    status = status &&
-      detail::intersect_bbox_bbox(m_min[i],
-                                  m_max[i],
-                                  otherBB.m_min[i],
-                                  otherBB.m_max[i]);
+    status &= detail::intersect_bbox_bbox(m_min[i],
+                                          m_max[i],
+                                          otherBB.m_min[i],
+                                          otherBB.m_max[i]);
```

Comment:
I definitely agree in spirit. However, the change to bitwise AND makes the traversal kernels ~20% faster on GPU, running mesh_tester with a 1M triangle car mesh:
 operation | Traversal Kernel 1 | Traversal Kernel 2
-------|-------------------|------------------
logical AND | 10.967ms | 23.732ms
bitwise AND | 9.145ms | 19.846ms
speedup | 1.199x | 1.195x

According to Godbolt (https://godbolt.org/z/WaGKxKfG3), the logical operation generates predicated branches to exit out when `status == false`, but my guess is that the branches are more expensive than running through all the computations as a consequence of warp divergence and/or data dependencies.

Reply:
Interesting and unexpected.
Do you have a sense of the effect on CPUs?

Reply:
Clarifying my previous comment: 
I'd like to retract my objection -- 20% is a significant improvement for this one change, so we should keep it.
It would still be worth knowing what the compromise is on the host side.

Reply:
It seems that this change leads to a slowdown within the traversal functions of roughly 7-8% on ppc64le. I wasn't able to measure the traversal independently on x86, but with regards to overall runtime there's a 10% hit with this change.

Would it be worth keeping the old method for the host side, guarded with `#ifdef __CUDA_ARCH__`?

Reply:
We have several primal operators with similar logic when operating in 2D and 3D 
(e.g. for intersections between rays, bboxes, segments and triangles, among others)

Presumably,  `status &= foo`  has the same efficiency as `status = status & foo`.

Could you wrap this in a (centralized) macro that performs  logical AND on the CPU and bitwise AND on the GPU?
or perhaps a `CombinationStatus` class with an overloaded AND operator if we want cleaner syntax?
(probably a better name than `CombinationStatus` though)

Reply:
This could be done in a separate PR though.  If so, please create an issue for this.

Reply:
Created an issue here: #582